src/is_intersection_empty.jl

```diff
@@ -418,3 +418,130 @@ function is_intersection_empty(H::Hyperplane{N},
                               )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
     return is_intersection_empty(Z, H, witness)
 end
+
+"""
+    is_intersection_empty(ls1::LineSegment{N},
+                          ls2::LineSegment{N},
+                          witness::Bool=false
+                         )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
+
+Check whether two line segments do not intersect, and otherwise optionally
+compute a witness.
+
+### Input
+
+- `ls1` -- first line segment
+- `ls2` -- second line segment
+- `witness` -- (optional, default: `false`) compute a witness if activated
+
+### Output
+
+* If `witness` option is deactivated: `true` iff ``ls1 ∩ ls2 = ∅``
+* If `witness` option is activated:
+  * `(true, [])` iff ``ls1 ∩ ls2 = ∅``
+  * `(false, v)` iff ``ls1 ∩ ls2 ≠ ∅`` and ``v ∈ ls1 ∩ ls2``
+
+### Algorithm
+
+The algorithm is inspired from [here](https://stackoverflow.com/a/565282), which
+again is the special 2D case of a 3D algorithm by R. Goldman, *Intersection of
+two lines in three-space*, 1990.
+
+We first check if the two line segments are parallel, and if so, if they are
+collinear.
+In the latter case, we check containment of any of the end points in the other
+line segment.
+Otherwise the lines are not parallel, so we can solve an equation of the
+intersection point, if it exists.
+
+### Notes
+
+Witness production is currently not supported.
```

Comment:
> Witness production is currently not supported.

but a witness is returned in some lines below. copy paste typo?

Reply:
Oh, I forgot to remove that. Thanks, I'll fix it now.